pkg/cloud/alibaba/assets/cloud-controller-manager-deployment.yaml

```diff
@@ -0,0 +1,108 @@
+      priorityClassName: system-cluster-critical
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: "kubernetes.io/hostname"
+              labelSelector:
+                matchLabels:
+                  app: alibaba-cloud-controller-manager
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
+          operator: Exists
+        - effect: NoExecute
+          key: node.kubernetes.io/unreachable
+          operator: Exists
+          tolerationSeconds: 120
+        - effect: NoExecute
+          key: node.kubernetes.io/not-ready
+          operator: Exists
+          tolerationSeconds: 120
+        - effect: NoSchedule
+          key: node.cloudprovider.kubernetes.io/uninitialized
+          operator: Exists
+        - effect: NoSchedule
+          key: node.kubernetes.io/not-ready
+          operator: Exists
+      containers:
+        - command:
+            - /bin/sh
+            - -c
+            - |
+              #!/bin/sh
+              set -o allexport
+              if [[ -f /etc/kubernetes/apiserver-url.env ]]; then
+                source /etc/kubernetes/apiserver-url.env
+              fi
+              exec /cloud-controller-manager \
+              --kubeconfig=/etc/kubernetes/cloud-controller-manager.conf \
+              --address=127.0.0.1 \
+              --allow-untagged-cloud=true \
+              --leader-elect=true \
+              --cloud-provider=alicloud \
+              --use-service-account-credentials=true \
+              --cloud-config=/etc/kubernetes/config/cloud-config.conf \
+              --feature-gates=ServiceNodeExclusion=true \
+              --configure-cloud-routes=false \
+              --allocate-node-cidrs=false
+          image: quay.io/openshift/origin-alibaba-cloud-controller-manager
+          livenessProbe:
+            failureThreshold: 8
+            httpGet:
+              host: 127.0.0.1
+              path: /healthz
+              port: 10258
+              scheme: HTTP
+            initialDelaySeconds: 15
+            timeoutSeconds: 15
+          name: cloud-controller-manager
+          resources:
+            requests:
+              cpu: 200m
+              memory: 50Mi
+          volumeMounts:
+            - mountPath: /etc/kubernetes/
+              name: k8s
+            - mountPath: /etc/ssl/certs
+              name: certs
+            - mountPath: /etc/pki
+              name: pki
+            - name: cloud-config
+              mountPath: /etc/kubernetes/config
+      volumes:
+        - hostPath:
+            path: /etc/kubernetes
+          name: k8s
+        - hostPath:
+            path: /etc/ssl/certs
+          name: certs
+        - hostPath:
+            path: /etc/pki
+          name: pki
+        - name: cloud-config
+          configMap:
```

Comment:
The name of the configmap created by the installer is cloud-conf.
```
          configMap:
            name: cloud-conf
            items:
              - key: cloud.conf
                path: cloud-config.conf
```

Reply:
got it.

Reply:
The configmap created by the installer in the ns 'openshift-config', but the ccm pod is in the ns 'openshift-cloud-controller-manager'. Can the configmap be mounted to the ccm pod?

Reply:
@kwoodson  I noticed that the config part of the cloud-provider-config.yaml file generated by the installer is similar to cloud-conf. Is there any connection between the two?

Reply:
This is part of the design of openshift, consumers of the configmap need to copy it to their own namespace. The CCCMO already handles this for you so you don't need to worry about it

Reply:
Ok, thanks,  But I am not sure if this helps to answer the two questions of @gujingit  above.

> Is the name of the configmap created by the installer "cloud-provider-config" and the key is "config" ? see https://github.com/openshift/installer/blob/2e99509e39d0aad420b7893a2454e619bdc9cab9/pkg/asset/manifests/cloudproviderconfig.go#L38-L52

> configMap:
>             name: cloud-conf
>             items:
>               - key: cloud.conf
>                 path: cloud-config.conf

I think the name here should be "cloud-conf" and the key should be "cloud.conf" of the configmap. Is my idea correct?

> The configmap created by the installer in the ns 'openshift-config', but the ccm pod is in the ns 'openshift-cloud-controller-manager'. Can the configmap be mounted to the ccm pod?

@JoelSpeed @kwoodson  Can you help me answer this question?

Reply:
> @JoelSpeed @kwoodson Can you help me answer this question?

This is what I was answering with the comment below:

> This is part of the design of openshift, consumers of the configmap need to copy it to their own namespace. The CCCMO already handles this for you so you don't need to worry about it

You can see we do this in Azure as an example https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/7d9a672c483f77ac788b1a52b944730889b347b4/pkg/cloud/azure/assets/cloud-controller-manager-deployment.yaml#L98-L103

Reply:
@JoelSpeed Thanks for the example here. 

I had to go back and read the documentation on configmaps and their documentation wasn't crystal clear.

@gujingit @bd233 
The `configMap.name` is the name of the configmap and it should be `cloud-conf`.  This is the copy that exists in the `openshift-cloud-controller-manager` namespace. The one the operator does for you.
```
oc edit cm -n openshift-cloud-controller-manager  cloud-conf 
configmap/cloud-conf edited
```

The `- key:` references the `configMap.items[0].key` in the `data` dictionary section of the configmap. The configmap looks like this:
```
apiVersion: v1
data:
  cloud.conf: "{\n\t\"Global\": {\n\t\t\"kubernetesClusterTag\": \"\",\n\t\t\"nodeMonitorPeriod\":
    0,\n\t\t\"nodeAddrSyncPeriod\": 0,\n\t\t\"uid\": \"\",\n\t\t\"vpcid\": \"\",\n\t\t\"region\":
    \"us-east-1\",\n\t\t\"zoneid\": \"\",\n\t\t\"vswitchid\": \"\",\n\t\t\"clusterID\":
    \"test-h44cl\",\n\t\t\"routeTableIDs\": \"\",\n\t\t\"serviceBackendType\": \"\",\n\t\t\"disablePublicSLB\":
    false,\n\t\t\"accessKeyID\": \"<redacted>\",\n\t\t\"accessKeySecret\":
    \"<redacted>\"\n\t}\n}\n"
kind: ConfigMap
metadata:
  creationTimestamp: "2021-09-02T16:46:04Z"
  name: cloud-conf
  namespace: openshift-cloud-controller-manager
  resourceVersion: "56192"
  uid: 47124e85-d887-4c70-b081-22db7cfbae31
```

The last `configMap.items[0].path` represents the mount path that exists on the filesystem. This value is what the file will show up as on the filesystem.  This value `--cloud-config=/etc/kubernetes/config/cloud-config.conf` from [here]( https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/94/files/b71a2f190d38485031e011e2e349aa06f9e9f9dc#diff-3385f0f98dcbbb6f0c5813829f456118119e9f5bcf8c76f8867b22fba9a725d6R65). The **`cloud-config.conf`** section is the `path`. As long as the values from the `--cloud-config` parameter to the `/cloud-controller-manager` container match this `path` then it should line up. 

Please follow the above information as well as the Azure example and I will test it tomorrow and verify it is all working properly.